docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,488 @@
+
+.. figure:: ../../data/how-to/hip_runtime_api/asynchronous/sequential_async_event.svg
+  :alt: Compare the different calls
+  :align: center
+
+The example codes
+
+.. tab-set::
+
+    .. tab-item:: Sequential
+
+      .. code-block:: cpp
+
+        #include <hip/hip_runtime.h>
+        #include <vector>
+        #include <iostream>
+
+        #define HIP_CHECK(expression)                \
+        {                                            \
+            const hipError_t status = expression;    \
+            if(status != hipSuccess){                \
+                    std::cerr << "HIP error "        \
+                        << status << ": "            \
+                        << hipGetErrorString(status) \
+                        << " at " << __FILE__ << ":" \
+                        << __LINE__ << std::endl;    \
+            }                                        \
+        }
+
+        // GPU Kernels
+        __global__ void kernelA(double* arrayA, size_t size){
+            const size_t x = threadIdx.x + blockDim.x * blockIdx.x;
+            if(x < size){arrayA[x] *= 2.0;}
+        };
+        __global__ void kernelB(double* arrayB, size_t size){
+            const size_t x = threadIdx.x + blockDim.x * blockIdx.x;
+            if(x < size){arrayB[x] += 3.0;}
+        };
+
+        int main()
+        {
+            constexpr int numOfBlocks = 256;
+            constexpr int threadsPerBlock = 4096;
+            constexpr int numberOfIterations = 50;
+            size_t arraySize = 1U << 20;
+
+            double *d_dataA;
+            double *d_dataB;
+            
+            std::vector<double> vectorA(arraySize, 4.0);
+            std::vector<double> vectorB(arraySize, 2.0);
+
+            // Allocate device memory
+            HIP_CHECK(hipMalloc(&d_dataA, arraySize * sizeof(*d_dataA)));
+            HIP_CHECK(hipMalloc(&d_dataB, arraySize * sizeof(*d_dataB)));
+
+            for(unsigned int iteration = 0; iteration < numberOfIterations; iteration++)
+            {
+                // Host to Device copies
+                HIP_CHECK(hipMemcpy(d_dataA, vectorA.data(), arraySize * sizeof(*d_dataA), hipMemcpyHostToDevice));
+                HIP_CHECK(hipMemcpy(d_dataB, vectorB.data(), arraySize * sizeof(*d_dataB), hipMemcpyHostToDevice));
+
+                // Launch the GPU kernels
+                hipLaunchKernelGGL(kernelA, dim3(numOfBlocks), dim3(threadsPerBlock), 0, 0, d_dataA, arraySize);
+                hipLaunchKernelGGL(kernelB, dim3(numOfBlocks), dim3(threadsPerBlock), 0, 0, d_dataB, arraySize);
+
+                // Device to Host copies
+                HIP_CHECK(hipMemcpy(vectorA.data(), d_dataA, arraySize * sizeof(*vectorA.data()), hipMemcpyDeviceToHost));
+                HIP_CHECK(hipMemcpy(vectorB.data(), d_dataB, arraySize * sizeof(*vectorB.data()), hipMemcpyDeviceToHost));
+            }
+
+            // Wait for all operations to complete
+            HIP_CHECK(hipDeviceSynchronize());
```

Comment:
Can we do some sort of validation here. All we did was see if kernels got executed and user did not get any error.

This should help user get confidence that the output is same regardless of sync or async.

Reply:
Made the requested changes.